### PR TITLE
Replace Slop with OptionParser

### PR DESCRIPTION
**Pull request by boniattirodrigo**
I forked the old pry-remote project and I'm trying to refactor it.

Here I replaced the Slop with OptionParser.

### CODE REVIEW

File: lib/pry-remote.rb [hunk 1]
```diff
@@ -260,26 +284,10 @@ def uri
...
-        on :f, "Disables loading of .pryrc and its plugins, requires, and command history "
-      end
-
-      exit if params.help?
+      params = Parser.parse(args)
 
-      @host = params[:server]
-      @port = params[:port]
+      @host = params[:server] || DefaultHost
+      @port = params[:port] || DefaultPort
```
Comment (boniattirodrigo) [inline lib/pry-remote.rb]: Parse class should define the default values. Let's improve this one in a follow-up PR.

File: lib/pry-remote.rb [hunk 2]
```diff
@@ -139,6 +139,30 @@ def input_proxy
     end
   end
 
+  class Parser
+    def self.parse(options)
+      args = {}
+
+      opt_parser = OptionParser.new do |opts|
+        opts.on("-sSERVER", "--server=SERVER", String, "Host of the server (#{DefaultHost})") do |s|
```
Comment (boniattirodrigo) [inline lib/pry-remote.rb:150]: I don't know how to define a default value. Can somebody help me with that?

Comment (jpechaves) [reply lib/pry-remote.rb:150]: I think you could set a default value into the `args = {}`. So when we run the new OptionParser initialization, if we don't pass any value as args, we can fall back into the default already defined in the args hash. Something like this:

```rb
args = {
  server: '127.0.0.1',
  port: '3000'
}
```

Comment (boniattirodrigo) [reply lib/pry-remote.rb:150]: Good call. I was looking for a way to set the default on the `opts.on` but it seems not possible.

**Pull request merged**